core/include/sched.h

```diff
@@ -188,6 +188,25 @@ void sched_register_cb(void (*callback)(uint32_t, uint32_t));
 
 #endif
 
+#ifdef BOARD_NATIVE
```

Comment:
Would you include the header on a condition or include it anyways and have the `#ifdef` inside the header. Should it be placed in the core folder as well?